openshift-kube-apiserver/admission/customresourcevalidation/imagetagmirrorset/validate_imagetagmirrorset.go

```diff
@@ -0,0 +1,109 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation/imagedigestmirrorset"
+)
+
+const PluginName = "config.openshift.io/ValidateImageTagMirrorSet"
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return newValidateITMS()
+	})
+}
+
+type validateCustomResourceWithClient struct {
+	admission.ValidationInterface
+
+	icspcli operatorsv1alpha1client.ImageContentSourcePoliciesGetter
+}
+
+func newValidateITMS() (admission.Interface, error) {
+	ret := &validateCustomResourceWithClient{}
+
+	delegate, err := customresourcevalidation.NewValidator(
+		map[schema.GroupResource]bool{
+			v1.Resource(imagedigestmirrorset.ItmsPlural): true,
+		},
+		map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+			v1.GroupVersion.WithKind(imagedigestmirrorset.ItmsKind): imagetagmirrorsetV1{icspcli: ret.getIcspcli},
+		})
+	if err != nil {
+		return nil, err
+	}
+	ret.ValidationInterface = delegate
+	return ret, nil
+}
+
+var _ admissionrestconfig.WantsRESTClientConfig = &validateCustomResourceWithClient{}
+
+func (i *validateCustomResourceWithClient) getIcspcli() operatorsv1alpha1client.ImageContentSourcePoliciesGetter {
+	return i.icspcli
+}
+
+func (i *validateCustomResourceWithClient) SetRESTClientConfig(restClientConfig rest.Config) {
+	var err error
+	i.icspcli, err = operatorsv1alpha1client.NewForConfig(&restClientConfig)
+	if err != nil {
+		utilruntime.HandleError(err)
+		return
+	}
+}
+
+func (i *validateCustomResourceWithClient) ValidateInitialization() error {
+	if i.icspcli == nil {
+		return fmt.Errorf(PluginName + " needs an imageContentSourcePoliciesGetter")
+	}
+	if initializationValidator, ok := i.ValidationInterface.(admission.InitializationValidator); ok {
+		return initializationValidator.ValidateInitialization()
+	}
+	return nil
+}
+
+type imagetagmirrorsetV1 struct {
+	icspcli func() operatorsv1alpha1client.ImageContentSourcePoliciesGetter
+}
+
+func (i imagetagmirrorsetV1) ValidateCreate(uncastObj runtime.Object) field.ErrorList {
+	return i.validateITMSUse("create")
+}
+
+func (i imagetagmirrorsetV1) ValidateUpdate(uncastObj runtime.Object, uncastOldObj runtime.Object) field.ErrorList {
+	return i.validateITMSUse("update")
+}
+
+func (i imagetagmirrorsetV1) ValidateStatusUpdate(uncastObj runtime.Object, uncastOldObj runtime.Object) field.ErrorList {
+	return i.validateITMSUse("update")
+}
+
+func (i imagetagmirrorsetV1) validateITMSUse(action string) field.ErrorList {
```

Comment:
This is duplicating the code from `imagedigestmirrorsetV1` with only slight modifications, can you rewrite it such that both `imagedigestmirrorsetV1` and `imagetagmirrorsetV1` will use the same logic, rather than implementing it twice?